exec.go

```diff
@@ -0,0 +1,77 @@
+package papergres
+
+import (
+	"time"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// Exec executes an ad-hoc query against a connection.
+// This is only recommended for use if you have a weird case where you need to
+// modify the connection string just for this query, like when creating
+// a new database. Otherwise just new New() and save the DAtabase instance.
+func Exec(sql string, conn Connection, args ...interface{}) *Result {
+	return conn.NewDatabase().Query(sql, args).Exec()
+}
+
+// ExecNonQuery executes an ad-hoc query against a connection.
+// This is only recommended for use if you have a weird case where you need to
+// modify the connection string just for this query, like when creating
+// a new database. Otherwise just new New() and save the Database instance.
+func ExecNonQuery(sql string, conn Connection, args ...interface{}) *Result {
+	return conn.NewDatabase().Query(sql, args).ExecNonQuery()
+}
+
+// execCommand is the single location that runs a command against the database (with the exception
+// of prepare statements).
+func execCommand(q *Query, cmd func(*Result) error, logArgs ...interface{}) *Result {
+	r := NewResult()
+
+	defer logQuery(q, r, time.Now(), logArgs...)
+
+	r.Err = cmd(r)
+	return r
+}
+
+// execDB creates the database for a command before passing it on to the execCommand function
+func execDB(q *Query, dbcmd func(*sqlx.DB, *Result) error) *Result {
```

Comment:
Not an interface since that's what we need here.
But, we can surely turn it into a function type to define the function signature.  Like this:
```go
// function type for a sqlx database command
type dbCmd func(*sqlx.DB, *Result) error
```
Thus the eventual function call would look like:
```go
func execDB(q *Query, dbcmd dbCmd) *Result { ... }
```

Good suggestion! Thanks, sent the updates in a new commit.